### DISCUSSION TITLE
tabBarItemHidden limitation when focused

### DISCUSSION THREAD

**Discussion by rochajulian** [callstack/react-native-bottom-tabs#208] [Q&A] [answered]
### Why it is needed?

Hey guys, I have a screen on my tab bar that I want to hide so I use the `tabBarItemHidden` prop. This works great until the route is focused because then the option doesn't hide the tab item anymore. This of course is specified in the [docs](https://okwasniewski.github.io/react-native-bottom-tabs/docs/guides/usage-with-react-navigation.html#tabbaritemhidden). My question is as follows: 

I read that this is a limitation on iOS. I am wondering if there is a clever workaround that you guys know of to avoid this behavior. In my app, and I can imagine in many others, we need the hidden tab bar item to remain hidden even if it gets focused.

**Comment by okwasniewski** [comment, marked as answer]
Hey, @rochajulian 

There are no plans to omit the native limitation. You can restructure your navigator, for example, using a stack navigator that embeds two routes: 1. A screen where you need to have a hidden bottom bar (for example, login) and then a second screen that hosts the bottom tab navigator. This should solve all of the needs for hiding a tab bar.
